iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/properties/ErrorListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser.properties;
+
+import javax.annotation.Nullable;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.sonar.api.batch.fs.TextPointer;
+import org.sonar.iac.common.extension.BasicTextPointer;
+import org.sonar.iac.common.extension.ParseException;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+
+public class ErrorListener extends BaseErrorListener {
+
+  @Nullable
+  private InputFileContext inputFileContext;
+
+  public ErrorListener(@Nullable InputFileContext inputFileContext) {
+    this.inputFileContext = inputFileContext;
+  }
+
+  @Override
+  public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
```

Comment:
Is it safe to ignore the exception here or can it contain additional information not present in `msg`?

Reply:
The default `ConsoleErrorListener` just prints `msg` and location into stderr. Here, the code throws the exception. I can add debug log, maybe it will be useful.

Reply:
I mean the parameter `RecognitionException e`, maybe it's more descriptive than `msg`. From the documentation it seems the opposite though, so feel free to do it as you see fit